src/components/WizardFormProvider/WizardFormProvider.tsx

```diff
@@ -1,31 +1,34 @@
 import { ReactNode, useState } from 'react';
 import FormScreens from 'constants/formScreens';
-import initialContextValue from 'constants/initialContextValue';
-import WizardFormContext from 'contexts/WizardFormContext';
-import PersonalDetailsValues from 'types/personalDetailsValues';
+import WizardFormContext, {
+  initialFormValues,
+} from 'contexts/WizardFormContext';
 
 interface Props {
   children: ReactNode;
 }
 
-interface FormValues {
-  formValues: { [FormScreens.PERSONAL_DETAILS]: PersonalDetailsValues };
-  setFormValues: (values: PersonalDetailsValues) => {
-    [FormScreens.PERSONAL_DETAILS]: PersonalDetailsValues;
-  };
-}
-
 export default function WizardFormProvider({ children }: Props) {
-  const [formValues, setFormValues] = useState<FormValues>(initialContextValue);
+  const [formValues, setFormValues] =
+    useState<typeof initialFormValues>(initialFormValues);
 
-  // eslint-disable-next-line react/jsx-no-constructed-context-values
-  const formDataStorage = {
-    formValues,
-    setFormValues,
+  const onSelectFormValues = <T extends FormScreens>(
```

Comment:
name this handler `handleSaveFormValues`. Pass it to context as `onSaveFormValues`